File: samples/sprite.cxx
#include <FK/FK.h>

using namespace std;
using namespace FK;
using namespace FK::Material;

int main(int, char *[])
{
	fk_AppWindow	window;
	fk_SpriteModel	sprite;
	fk_Block		block(1.0, 1.0, 1.0);
	fk_Model		model;
	fk_Vector		origin(0.0, 0.0, 0.0);
	fk_RectTexture	rect;
	fk_Model		model2;
	int				count;
	std::string		str;

	fk_System::setcwd();

	sprite.initFont("rm1b.ttf");
	sprite.setPositionLT(0.0, 0.0);
	window.entry(sprite);

	model.setShape(&block);
	model.glMoveTo(0.0, 6.0, 0.0);
	model.setMaterial(Yellow);
	window.entry(model);

	window.setCameraPos(0.0, 5.0, 20.0);
	window.setCameraFocus(0.0, 5.0, 0.0);
	window.setSize(800, 600);
	window.setBGColor(0.6, 0.7, 0.8);
	window.open();
	window.showGuide(FKUT_GRID_XZ);
	rect.setImage(sprite.getImage());
	rect.setTextureSize(6.0, 2.0);
	model2.setShape(&rect);
	model2.glMoveTo(0.0, 2.0, 0.0);
	window.entry(&model2);

	count = 0;
	for(count = 0; window.update() == true; count++) {
		str = "count = " + std::to_string(count);
		sprite.drawText(str, true);
		sprite.setPositionLT(0.0, 0.0);
		model.glRotateWithVec(origin, fk_Y, FK_PI/360.0);
		model2.glRotateWithVec(origin, fk_Y, FK_PI/500.0);
		rect.setImage(sprite.getImage());
		if(count == 1000) {
			sprite.getImage()->writePNG("sample.png");
			fk_Window::printf("1000");
		}
	}
	return 0;
}

File: CLI/Test/FK_CLI_Sprite/Sprite.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using FK_CLI;

namespace FK_CLI_Sprite
{
	class Program
	{
		static void Main(string[] args)
		{
			var window = new fk_AppWindow();
			fk_Material.InitDefault();
			
			var sprite = new fk_SpriteModel();
			var block = new fk_Block(1.0, 1.0, 1.0);
			var model = new fk_Model();
			var origin = new fk_Vector(0.0, 0.0, 0.0);
			int count;
			string str, space;
 
			if(sprite.InitFont("mona.ttf") == false) {
				System.Console.WriteLine("Font Error");
			}

			sprite.SetPositionLT(-240.0, 230.0);

			sprite.Text.MonospaceMode = true;
			sprite.Text.MonospaceSize = 12;
			window.Entry(sprite);
 
			model.Shape = block;
			model.GlMoveTo(0.0, 6.0, 0.0);
			model.Material = fk_Material.Yellow;
			window.Entry(model);
			window.CameraPos = new fk_Vector(0.0, 5.0, 20.0);
			window.CameraFocus = new fk_Vector(0.0, 5.0, 0.0);
			window.Size = new fk_Dimension(800, 600);
			window.BGColor = new fk_Color(0.6, 0.7, 0.8);
			window.Open();
			window.ShowGuide(fk_GuideMode.GRID_XZ);
 
			count = 0;
			while(window.Update() == true) {
                if (count < 10) space = "   ";
				else if(count < 100) space = "  ";
				else if(count < 1000) space = " ";
				else space = "";
				str = "count = " + space + count.ToString();
				sprite.DrawText(str, true);
				model.GlRotateWithVec(origin, fk_Axis.Y, Math.PI/360.0);
				count++;
			}
		}
	}
}
